File: src/main/java/eu/execom/fabut/AbstractFabutTest.java
package eu.execom.fabut;

import org.junit.After;
import org.junit.Before;

/**
 * TODO add comments.
 * 
 * @author Dusko Vesin
 * 
 */
public abstract class AbstractFabutTest extends Fabut implements IFabutTest {

    public AbstractFabutTest() {
        super();
    }

    @Before
    @Override
    public void fabutBeforeTest() {
        Fabut.beforeTest(this);
    };

    @Override
    @After
    public void fabutAfterTest() {
        Fabut.afterTest();
    }

}

File: src/main/scala/eu/execom/fabut/AbstractFabutTest.scala
package eu.execom.fabut

import org.junit.{After, Before}

trait AbstractFabutTest extends Fabut with IFabutTest {

  @Before
  override def beforeTest(): Unit = {
    Fabut.beforeTest(this)
  }

  @After
  override def afterTest(): Unit = {
    Fabut.afterTest()
  }
}
